tasks/managed/rh-sign-image-cosign/rh-sign-image-cosign.yaml

```diff
@@ -73,37 +87,84 @@ spec:
-                  --rekor-url="$REKOR_URL"
-                  --key
-                  "$SIGN_KEY"
-                )
-            else
-                COSIGN_COMMON_ARGS=(
-                  --tlog-upload=false
-                  --key
-                  "$SIGN_KEY"
-                )
-            fi
-            echo "Signing manifest $1 ($2)"
             attempt=0
+            backoff1=2
+            backoff2=3
             until [ "$attempt" -gt "$(params.retries)" ] ; do # 3 retries by default
-                cosign -t 3m0s sign\
-                "${COSIGN_COMMON_ARGS[@]}" \
-                --sign-container-identity "$2"\
-                "$1" && break
+                cosign "$@" && break
+                sleep $backoff2
+
+                # Fibbonaci backoff
+                old_backoff1=$backoff1
+                backoff1=$backoff2
+                backoff2=$((old_backoff1 + backoff2))
                 attempt=$((attempt+1))
             done
             if [ "$attempt" -gt "$(params.retries)" ] ; then
-              echo "Max retries exceeded."
+              echopid "Max retries exceeded."
               exit 1
             fi
         }
+        function check_existing_signatures() {
+          local identity=$1
+          local reference=$2
+          local digest=$3
+          if [ -v REKOR_URL ]; then
+              COSIGN_REKOR_ARGS="--rekor-url=$REKOR_URL"
+              REKOR_PUBLIC_KEY_FILE=$(mktemp)
+              echo -n "$REKOR_PUBLIC_KEY" > "$REKOR_PUBLIC_KEY_FILE"
+              export SIGSTORE_REKOR_PUBLIC_KEY="$REKOR_PUBLIC_KEY_FILE"
+          else
+              COSIGN_REKOR_ARGS="--insecure-ignore-tlog=true"
+          fi
+          verify_output=$(run_cosign verify "${COSIGN_REKOR_ARGS[@]}" --key "$PUBLIC_KEY_FILE" "$reference")
+          found_signatures=$(echo "$verify_output" | jq -j '['\
+        '.[]|select(.critical.image."docker-manifest-digest"| contains("'"$digest"'"))'\
+        '|select(.critical.identity."docker-reference"| contains("'"$identity"'"))'\
+        ']|length')
+          echo "$found_signatures"
+        }
+        function check_and_sign() {
+          local identity=$1
+          local reference=$2
+          local digest=$3
+          found_signatures=$(check_existing_signatures "$identity" "$reference@$digest" "$digest")
+          if [ -z "$found_signatures" ]; then
+            found_signatures=0
+          fi
+          echopid "FOUND SIGNATURES for ${identity} ${digest}: $found_signatures"
+
+          if [ -v REKOR_URL ]; then
+              COSIGN_REKOR_ARGS="-y --rekor-url=$REKOR_URL"
+              REKOR_PUBLIC_KEY_FILE=$(mktemp)
+              echo -n "$REKOR_PUBLIC_KEY" > "$REKOR_PUBLIC_KEY_FILE"
+              export SIGSTORE_REKOR_PUBLIC_KEY="$REKOR_PUBLIC_KEY_FILE"
+          else
+              COSIGN_REKOR_ARGS="--tlog-upload=false"
+          fi
+
+          if [ "$found_signatures" -eq 0 ]; then
+            run_cosign -t 3m0s sign "${COSIGN_REKOR_ARGS[@]}" \
```

Comment:
```
$ COSIGN_REKOR_ARGS="-y --rekor-url=abc"; cosign sign "${COSIGN_REKOR_ARGS[@]}"
WARNING: the -y --rekor-url=abc flag is deprecated and will be removed in a future release. Please use the --y --rekor-url=abc flag instead.
Error: unknown flag: --y --rekor-url
main.go:74: error during command execution: unknown flag: --y --rekor-url
```
Remove the quotes?

Reply:
The quotes were used because otherwise shellcheck complains. 

But as I suggested in https://github.com/konflux-ci/release-service-catalog/pull/771#discussion_r1919722671 , an array should be used. Currently COSIGN_REKOR_ARGS is defined as a plain string. See here for an example: https://github.com/konflux-ci/release-service-catalog/blob/development/tasks/managed/create-pyxis-image/create-pyxis-image.yaml#L165

Reply:
@emilyzheng @mmalina can we resolve this? It is quite old

Reply:
It doesn't seem like the issue has been addressed. Ping @midnightercz

Reply:
This is already fixed, on line 146 declare -a COSIGN_REKOR_ARGS=()